tests/unit/cunumeric/test_patch.py

```diff
@@ -0,0 +1,31 @@
+# Copyright 2021-2022 NVIDIA Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import os
+from subprocess import run
+
+legate = os.environ["_"]
```

Comment:
When I run legate, this env var has the absolute path of the `legate` executable:
```
>>> os.environ["_"]
'/home/bryan/work/cunumeric/../legate.core/install37/bin/legate'
```

Is this documented and/or guaranteed to persist in the future? These tests do need process isolation so so means of executing `legate` is needed (`sys.executable` does not help here)

Reply:
AFAIU `_` is a special variable that is guaranteed to be set if `legate` is invoked through `bash`, see https://tldp.org/LDP/Bash-Beginners-Guide/html/sect_03_02.html. No idea if other shells provide similar guarantees.

If we don't like this, then we could instead directly build a command line for the actual `legion_python` executable (that's pretty much what the `bin/legate` wrapper script does). Try running the original testing script with `legate --verbose -c '...'`, and you'll see a printout of the `legion_python` invocation.

`sys.executable` should always contain an absolute path to the `legion_python` executable.

Reply:
@manopapad that definitely does not seem to be the case for `sys.executable` when I try:
```
legate37 ❯ legate
WARNING: Disabling control replication for interactive run
Welcome to Legion Python interactive console
>>> import sys
>>> sys.executable
'/home/bryan/anaconda3/envs/legate37/bin/python3'
```
I will try to emulate the `legate` wrapper, that  seems like it would be the least fragile

Reply:
OK, I also see different behavior regarding `sys.executable` on different systems. Sometimes it shows `legion_python` and sometimes it shows `python3`. If that's the case then I don't immediately see a (guaranteed) easy way to get a path to the `legion_python` executable or the `legate` wrapper script. We could potentially set a well-known envvar or global variable in the top-most invocation of `legate`.

Reply:
Perhaps I should mark these two tests `@skip` for now